Build error with static Linux SDK.

**Describe the bug**
```console
➜  swift-subprocess git:(main) ✗ swift build --swift-sdk x86_64-swift-linux-musl -c release
[1/1] Planning build
Building for production...
/home/zane/.scribe/Projects/swift-subprocess/Sources/_SubprocessCShims/process_shims.c:257:42: error: token is not a valid binary operator in a preprocessor subexpression
  257 | #if defined(__GLIBC__) && !__GLIBC_PREREQ(2, 29)
      |                           ~~~~~~~~~~~~~~~^
/home/zane/.scribe/Projects/swift-subprocess/Sources/_SubprocessCShims/process_shims.c:287:42: error: token is not a valid binary operator in a preprocessor subexpression
  287 | #if defined(__GLIBC__) && !__GLIBC_PREREQ(2, 29)
      |                           ~~~~~~~~~~~~~~~^
2 errors generated.
[2/4] Write swift-version-74104DCF1D77AF14.txt
```

**To Reproduce**
Steps to reproduce the behavior:
Clone subprocess repo.
Install Swift 6.1 (I used swiftly)
[install static swift SDK](https://www.swift.org/documentation/articles/static-linux-getting-started.html)
swift build --swift-sdk x86_64-swift-linux-musl -c release

**Expected behavior**
Should build.

**Environment (please complete the following information):**
```console
➜  swift-subprocess git:(main) ✗ ldd --version
ldd (Ubuntu GLIBC 2.39-0ubuntu8.4) 2.39
Copyright (C) 2024 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
Written by Roland McGrath and Ulrich Drepper.
➜  swift-subprocess git:(main) ✗ swift --version
Swift version 6.1 (swift-6.1-RELEASE)
Target: x86_64-unknown-linux-gnu
➜  swift-subprocess git:(main) ✗ swift sdk list
swift-6.1-RELEASE_static-linux-0.0.1
```

```console
PRETTY_NAME="Ubuntu 24.04.2 LTS"
NAME="Ubuntu"
VERSION_ID="24.04"
VERSION="24.04.2 LTS (Noble Numbat)"
VERSION_CODENAME=noble
ID=ubuntu
ID_LIKE=debian
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
UBUNTU_CODENAME=noble
LOGO=ubuntu-logo
```

**Additional context**

--- a/Sources/Subprocess/Platforms/Subprocess+Linux.swift
+++ b/Sources/Subprocess/Platforms/Subprocess+Linux.swift
@@ -257,6 +257,24 @@ private let _childProcessContinuations:
 
 private let signalSource: SendableSourceSignal = SendableSourceSignal()
 
+private extension siginfo_t {
+    var si_status: Int32 {
+        #if canImport(Glibc)
+        return _sifields._sigchld.si_status
+        #elseif canImport(Musl)
+        return __si_fields.__si_common.__second.__sigchld.si_status
+        #endif
+    }
+
+    var si_pid: pid_t {
+        #if canImport(Glibc)
+        return _sifields._sigchld.si_pid
+        #elseif canImport(Musl)
+        return __si_fields.__si_common.__first.__piduid.si_pid
+        #endif
+    }
+}
+
 private let setup: () = {
     signalSource.setEventHandler {
         _childProcessContinuations.withLock { continuations in
@@ -268,9 +286,9 @@ private let setup: () = {
                 var status: TerminationStatus? = nil
                 switch siginfo.si_code {
                 case .init(CLD_EXITED):
-                    status = .exited(siginfo._sifields._sigchld.si_status)
+                    status = .exited(siginfo.si_status)
                 case .init(CLD_KILLED), .init(CLD_DUMPED):
-                    status = .unhandledException(siginfo._sifields._sigchld.si_status)
+                    status = .unhandledException(siginfo.si_status)
                 case .init(CLD_TRAPPED), .init(CLD_STOPPED), .init(CLD_CONTINUED):
                     // Ignore these signals because they are not related to
                     // process exiting
@@ -279,7 +297,7 @@ private let setup: () = {
                     fatalError("Unexpected exit status: \(siginfo.si_code)")
                 }
                 if let status = status {
-                    let pid = siginfo._sifields._sigchld.si_pid
+                    let pid = siginfo.si_pid
                     if let existing = continuations.removeValue(forKey: pid),
                         case .continuation(let c) = existing
                     {

--- a/Sources/_SubprocessCShims/process_shims.c
+++ b/Sources/_SubprocessCShims/process_shims.c
@@ -251,6 +251,9 @@ int _subprocess_spawn(
 
 // MARK: - Linux (fork/exec + posix_spawn fallback)
 #if TARGET_OS_LINUX
+#ifndef __GLIBC_PREREQ
+#define __GLIBC_PREREQ(maj, min) 0
+#endif
 
 #if _POSIX_SPAWN
 static int _subprocess_is_addchdir_np_available() {

